src/lib/functions/netlify-function.mjs

```diff
@@ -93,11 +103,22 @@ export default class NetlifyFunction {
       const srcFilesDiff = this.getSrcFilesDiff(srcFilesSet)
 
       this.buildData = buildData
+      this.buildError = null
       this.srcFiles = srcFilesSet
       this.schedule = schedule || this.schedule
 
+      if (!this.isSupported()) {
+        throw new Error(
+          `Function requires Node.js version ${V2_MIN_NODE_VERSION} or above, but ${nodeVersion.slice(
+            1,
+          )} is installed. Refer to https://ntl.fyi/functions-node18 for information on how to update.`,
```

Comment:
This will eventually point to a forums post where we explain why this requirement exists and how to update Node.js.